File: src/main/java/fr/epicanard/mapsaver/models/config/Options.java
package fr.epicanard.mapsaver.models.config;

import java.util.Optional;

public class Options {
    public Integer PageSize = 10;

    public void verifyPagination() {
        this.PageSize = Optional.ofNullable(this.PageSize).filter(s -> s >= 1).orElse(10);
    }
}

File: src/main/scala/fr/epicanard/mapsaver/resources/config/Options.scala
package fr.epicanard.mapsaver.resources.config

import fr.epicanard.mapsaver.circe.CapitalizeConfiguration
import fr.epicanard.mapsaver.models.map.Visibility
import io.circe.generic.extras.ConfiguredJsonCodec

@ConfiguredJsonCodec
case class Options(
    pageSize: Int,
    defaultVisibility: Visibility,
    consumeEmptyMap: Boolean
)

object Options extends CapitalizeConfiguration
